src/Service.Tests/GraphQLBuilder/MutationBuilderTests.cs

```diff
@@ -1012,34 +1012,49 @@ public static ObjectTypeDefinitionNode GetMutationNode(DocumentNode mutationRoot
+        [DataRow(GraphQLTestHelpers.PERSON_GQL, new string[] { "Person" }, null, null, new string[] { "Person" }, null, false,
+            DisplayName = "Mutation name and description validation for singular entity name with singular plural not defined - Multiple Create Operation disabled")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL, new string[] { "Person" }, new string[] { "Person" }, new string[] { "People" }, new string[] { "Person" }, new string[] { "People" }, true,
+            DisplayName = "Mutation name and description validation for singular entity name with singular plural defined - Multiple Create Operation enabled")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL, new string[] { "Person" }, new string[] { "Person" }, new string[] { "People" }, new string[] { "Person" }, null, false,
+            DisplayName = "Mutation name and description validation for singular entity name with singular plural defined - Multiple Create Operation disabled")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL + GraphQLTestHelpers.BOOK_GQL, new string[] { "Person", "Book" }, new string[] { "Person", "Book" }, new string[] { "People", "Books" }, new string[] { "Person", "Book" }, new string[] { "People", "Books" }, true,
+            DisplayName = "Mutation name and description validation for multiple entities with singular, plural - Multiple Create Operation enabled")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL + GraphQLTestHelpers.BOOK_GQL, new string[] { "Person", "Book" }, new string[] { "Person", "Book" }, new string[] { "People", "Books" }, new string[] { "Person", "Book" }, null, false,
+            DisplayName = "Mutation name and description validation for multiple entities with singular, plural - Multiple Create Operation disabled")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL + GraphQLTestHelpers.BOOK_GQL, new string[] { "Person", "Book" }, null, null, new string[] { "Person", "Book" }, new string[] { "People", "Books" }, true,
+            DisplayName = "Mutation name and description validation for multiple entities with singular plural not defined - Multiple Create Operation enabled")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL + GraphQLTestHelpers.BOOK_GQL, new string[] { "Person", "Book" }, null, null, new string[] { "Person", "Book" }, null, false,
+            DisplayName = "Mutation name and description validation for multiple entities with singular plural not defined - Multiple Create Operation disabled")]
         public void ValidateMutationsAreCreatedWithRightName(
             string gql,
             string[] entityNames,
             string[] singularNames,
             string[] pluralNames,
-            string[] expectedNames
+            string[] expectedNames,
+            string[] expectedCreateMultipleMutationNames,
+            bool isMultipleCreateOperationEnabled
```

Comment:
i understand there would be code duplication if you had a separate test was explicitly for multiple mutations. The benefits outweigh the code duplication in my opinion:
- easer to read the DataRows with way fewer parameters
- fewer conditionals in test logic that make this harder to follow
- don't have to confuse test logic that explicitly tests point mutation schema creation.

Reply:
I considered these factors but somehow I was leaning more towards performing validations for scenarios with feature flag on/off in the same test.

I agree with the reasoning - I've split this in two different tests. One for validating scenarios with multiple create enabled, the other with multiple create disabled.